debezium-connector-mysql/src/main/java/io/debezium/connector/mysql/MySqlHistoryRecordComparator.java

```diff
@@ -145,4 +143,79 @@ else if (recordedGtidSetStr != null) {
         // The binlog coordinates are the same ...
         return true;
     }
+
+    public static class BinlogFilename implements Comparable<BinlogFilename> {
+        final private String baseFilename;
+        final private long extension;
+
+        private BinlogFilename(String baseFilename, long extension) {
+            this.baseFilename = baseFilename;
+            this.extension = extension;
+        }
+
+        public static BinlogFilename of(String filename) {
+            if (!filename.contains(".")) {
+                throw new IllegalArgumentException("Invalid binlog filename format: " + filename);
```

Comment:
I'd use a more specific error message regarding what exactly is invalid about the filename:

```suggestion
                throw new IllegalArgumentException("Filename does not have an extension: " + filename);
```